File: matlab/Unidad_2/ej13.m
% Realice procesamientos que involucren 
% operaciones aritméticas y booleanas.

[file, path] = uigetfile({'*.tiff';'*.JPG';'*.bmp';'*.jpg'});
dir = strcat(path,file);
info = imfinfo(dir);

switch info.ColorType
    case 'grayscale'
        I1=imread(dir);
    case 'truecolor'
        rgb=imread(dir);
        I1=rgb2gray(rgb);
    case 'indexed'
        [X,map]=imread(dir);
        I1=ind2gray(X,map);
end

[file, path] = uigetfile({'*.tiff';'*.JPG';'*.bmp';'*.jpg'});
dir = strcat(path,file);
info = imfinfo(dir);

switch info.ColorType
    case 'grayscale'
        I2=imread(dir);
    case 'truecolor'
        rgb=imread(dir);
        I2=rgb2gray(rgb);
    case 'indexed'
        [X,map]=imread(dir);
        I2=ind2gray(X,map);
end


% Binarize I1
th1 = graythresh(I1);
BW1 = imbinarize(I1, th1);

% Binarize I2
th2 = graythresh(I2);
BW2 = imbinarize(I2, th2);

% Logic Operations
R1 = not(BW2) ;
R2 = and(R1, I1); % Este no funca

% Plots
subplot(2,3,1); imshow(I1);
title('Imagen original 1');

subplot(2,3,2); imshow(I2);
title('Imagen original 2');

subplot(2,3,3); imshow(BW1);
title('Imagen 1 binarizada');

subplot(2,3,4); imshow(BW2);
title('Imagen 2 binarizada');

subplot(2,3,5); imshow(R1);
title('not Imagen 2');

subplot(2,3,6); imshow(R2);
title('(not Imagen 2) or Imagen 1');

File: python/Unidad_2/ej13.py
# -*- coding: utf-8 -*-
"""
Created on Sun Apr 25 18:07:44 2021

@author: Javier
"""

# Realice procesamientos que involucren operaciones aritméticas y booleanas.
# NO ESTA TERMINADO, ESTOY VIENDO LA POSIBILIDAD DE USAR OPENCV
# https://answers.opencv.org/question/228538/how-to-create-a-binary-mask-for-medical-images/


from matplotlib import pyplot as plt
from skimage.io import imread
from skimage.color import rgb2gray
from skimage.filters import threshold_otsu as th_otsu
import tkinter as tk
from tkinter import filedialog
    
def file_read():
    file_path = filedialog.askopenfilename()
    return file_path

try:
    root = tk.Tk()
    root.withdraw() 
    image = imread(file_read()) # Leo imagen
    if len(image.shape) == 3: # Si es Truecolor, paso a grayscale
        img_gray = rgb2gray(image)
    else: # Sino, solo paso los valores
        img_gray = image
    val = th_otsu(img_gray, nbins=256)
    mask = img_gray > val
    masked_image = img_gray * mask
    
    fig, (ax0, ax1) = plt.subplots(nrows=1, ncols=2)
    ax0.imshow(image, cmap=plt.cm.gray)
    ax1.imshow(masked_image, cmap=plt.cm.gray)
    
except:
    print('No abriste ninguna imagen!')


print('👋🏽')
root.destroy()